File: src/main/java/com/drunkensoftball/api/user/domain/UserRequest.java
package com.drunkensoftball.api.user.domain;


public class UserRequest {

    private String username;
    private String authenticationType;
    private String email;

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getAuthenticationType() {
        return authenticationType;
    }

    public void setAuthenticationType(String authenticationType) {
        this.authenticationType = authenticationType;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
}

File: src/main/kotlin/com/drunkensoftball/api/user/domain/UserRequest.kt
package com.drunkensoftball.api.user.domain

import org.hibernate.validator.constraints.Email
import org.hibernate.validator.constraints.NotBlank
import javax.validation.constraints.Size


class UserRequest {

    @get: NotBlank(message = "{username.required}")
    var username: String = ""

    @get: NotBlank(message = "{password.required}")
    @get: Size(min=6, message = "{password.size}")
    var password: String = ""

    @get: NotBlank(message = "{email.required}")
    @get: Email(message = "{email.invalid}")
    var email: String = ""

    var firstName: String? = null
    var lastName: String? = null
}
